### PR TITLE
[Merged by Bors] - refactor: flip `LinearMap.convexHull_image` and rename to `LinearMap.image_convexHull`

### PR DESCRIPTION
**Pull request by grunweg**
- flip `LinearMap.convexHull_image` and rename to `image_convexHull`
- while at it, also flip the direction of `convexHull_smul` and `convexHull_neg` to match this

- fix argument order of `AffineMap.convexHull_image`: have the `AffineMap` argument come first; there's no good reason not to and this enables dot notation
- inline variable `(s : Set E)` to achieve this; this is slightly clearer anyway

[zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/image_convexHull.20vs.20convexHull_image)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — COMMENTED** [review]
As stated on Zulip, the other direction is preferable to match other lemmas about maps of constructs along morphisms.

File: Mathlib/Analysis/Convex/Hull.lean [hunk 1]
```diff
@@ -182,7 +182,7 @@ section OrderedCommSemiring
 variable [OrderedCommSemiring 𝕜] [AddCommMonoid E] [Module 𝕜 E]
 
 theorem convexHull_smul (a : 𝕜) (s : Set E) : convexHull 𝕜 (a • s) = a • convexHull 𝕜 s :=
```
Comment (grunweg) [inline Mathlib/Analysis/Convex/Hull.lean:184]: Should this lemma also be flipped?

File: Mathlib/Analysis/Convex/Hull.lean [hunk 2]
```diff
@@ -193,33 +193,33 @@ variable [OrderedRing 𝕜]
...
 
 @[simp]
-theorem affineSpan_convexHull : affineSpan 𝕜 (convexHull 𝕜 s) = affineSpan 𝕜 s := by
+theorem affineSpan_convexHull (s : Set E) : affineSpan 𝕜 (convexHull 𝕜 s) = affineSpan 𝕜 s := by
   refine' le_antisymm _ (affineSpan_mono 𝕜 (subset_convexHull 𝕜 s))
   rw [affineSpan_le]
   exact convexHull_subset_affineSpan s
 #align affine_span_convex_hull affineSpan_convexHull
 
 theorem convexHull_neg (s : Set E) : convexHull 𝕜 (-s) = -convexHull 𝕜 s := by
```
[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Analysis/Convex/Hull.lean]: Flipped now.

File: Mathlib/Analysis/Convex/Hull.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Analysis/Convex/Hull.lean]: Flipped now.

File: Mathlib/Analysis/Complex/Convex.lean [hunk 3]
```diff
@@ -21,7 +21,7 @@ lemma convexHull_reProdIm (s t : Set ℝ) :
   calc
     convexHull ℝ (equivRealProdLm ⁻¹' (s ×ˢ t)) = equivRealProdLm ⁻¹' (convexHull ℝ (s ×ˢ t)) := by
       simpa only [← LinearEquiv.image_symm_eq_preimage]
-        using equivRealProdLm.symm.toLinearMap.image_convexHull (s ×ˢ t)
+        using ((equivRealProdLm.symm.toLinearMap).image_convexHull (s ×ˢ t)).symm
```
[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Analysis/Complex/Convex.lean:24]: That does not build:
```
type mismatch
  s ×ˢ t
has type
  Set (ℝ × ℝ) : outParam Type
but is expected to have type
  Set ℂ : Type```

[the remark this answers is not in the document]
Comment (YaelDillies) [reply Mathlib/Analysis/Complex/Convex.lean:24]: Ah sorry. What if you replace the `s x t` argument by an underscore?

[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Analysis/Complex/Convex.lean:24]:
```
type mismatch
  h✝
has type
  ⇑↑equivRealProdLm '' (convexHull ℝ) ?m.5210 = (convexHull ℝ) (⇑↑equivRealProdLm '' ?m.5210) : Prop
but is expected to have type
  (convexHull ℝ) (⇑(LinearEquiv.symm equivRealProdLm) '' s ×ˢ t) =
    ⇑(LinearEquiv.symm equivRealProdLm) '' (convexHull ℝ) (s ×ˢ t) : PropLean 4

[the remark this answers is not in the document]
Comment (grunweg) [reply Mathlib/Analysis/Complex/Convex.lean:24]: Feel free to tinker with and push a tweak yourself!

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**